src/core/rest/createRestRoutes/createRestRoutes.ts

```diff
@@ -60,40 +59,68 @@ export const createRestRoutes = ({
+              ) {
+                return resolveEntityValues({
+                  checkMode: topLevelConvertedDescriptor.checkMode,
+                  actualValue
+                });
+              }
+
+              return resolveEntityValues({
+                checkMode: topLevelConvertedDescriptor.checkMode,
+                actualValue,
+                descriptorValue: topLevelConvertedDescriptor.value,
+                oneOf: topLevelConvertedDescriptor.oneOf as true | false
+              });
             }
 
             const isEntityBodyByTopLevelArray =
               entityName === 'body' && Array.isArray(entityDescriptorOrValue);
             if (isEntityBodyByTopLevelArray) {
-              return entityDescriptorOrValue.some((entityDescriptorOrValueElement) =>
-                // ✅ important:
-                // bodyParser sets body to empty object if body not sent or invalid, so assume {} as undefined
-                resolveEntityValues(
-                  checkMode,
-                  Object.keys(request.body).length ? request.body : undefined,
-                  entityDescriptorOrValueElement
-                )
-              );
+              // ✅ important:
+              // bodyParser sets body to empty object if body not sent or invalid, so assume {} as undefined
+              const actualValue = Object.keys(request.body).length ? request.body : undefined;
+              return resolveEntityValues({
+                checkMode: 'equals',
+                actualValue,
+                descriptorValue: entityDescriptorOrValue
+              });
             }
 
             const recordOrArrayEntries = Object.entries(entityDescriptorOrValue) as Entries<
               Exclude<RestEntity, TopLevelPlainEntityDescriptor | TopLevelPlainEntityArray>
             >;
-            return recordOrArrayEntries.every(([entityKey, mappedEntityDescriptor]) => {
-              const { checkMode, value: descriptorValue } =
-                convertToEntityDescriptor(mappedEntityDescriptor);
-              const flattenEntity = flatten<any, any>(request[entityName]);
-              // ✅ important: transform header keys to lower case because browsers send headers in lowercase
-              return resolveEntityValues(
-                checkMode,
-                flattenEntity[entityName === 'headers' ? entityKey.toLowerCase() : entityKey],
-                descriptorValue
+            return recordOrArrayEntries.every(([entityKey, mappedEntityDescriptorOrValue]) => {
+              const propertyLevelConvertedDescriptor = convertToEntityDescriptor(
+                mappedEntityDescriptorOrValue
               );
+              const actualEntity = flatten<any, any>(request[entityName]);
+
+              // ✅ important: transform header keys to lower case because browsers send headers in lowercase
+              const actualValue =
+                actualEntity[entityName === 'headers' ? entityKey.toLowerCase() : entityKey];
```

Comment:
// ✅ important: transform header keys to lower case because browsers send headers in lowercase
cosnt actualKey = entityName === 'headers' ? entityKey.toLowerCase() : entityKey
const actualValue = actualEntity[actualKey]